pym/portage/meta-manifest.py

```diff
@@ -0,0 +1,185 @@
+
+from __future__ import unicode_literals
+
+import argparse
+import errno
+import io
+import logging
+import re
+import stat
+import subprocess
+import sys
+import warnings
+
+import portage
+portage.proxy.lazyimport.lazyimport(globals(),
+        'portage.checksum:hashfunc_map,perform_multiple_checksums,' + \
+                'verify_all,_apply_hash_filter,_filter_unaccelarated_hashes',
+        'portage.repository.config:_find_invalid_path_char',
+        'portage.util:write_atomic,writemsg_level',
+)
+
+from portage import os
+from portage import _encodings
+from portage import _unicode_decode
+from portage import _unicode_encode
+import manifest
+from portage.exception import DigestException, FileNotFound, \
+        InvalidDataType, MissingParameter, PermissionDenied, \
+        PortageException, PortagePackageException
+from const import (MANIFEST1_HASH_FUNCTIONS, MANIFEST2_HASH_DEFAULTS,
+        MANIFEST2_HASH_FUNCTIONS, MANIFEST2_IDENTIFIERS, MANIFEST2_REQUIRED_HASH)
+from portage.localization import _
+
+if sys.hexversion >= 0x3000000:
+	# pylint: disable=W0622
+	_unicode = str
+	basestring = str
+else:
+	_unicode = unicode
+
+def guessManifestFileType(filename):
+        """ Perform a best effort guess of which type the given filename is, avoid using this if possible """
+        if filename.startswith("files" + os.sep + "digest-"):
+                return None
+        elif filename.startswith("files" + os.sep):
+                return "AUX"
+        elif filename.endswith(".ebuild"):
+                return "EBUILD"
+        elif filename == "Manifest" or filename == "MetaManifest":
+                return "MANIFEST"
+        elif filename.endswith(".eclass"):
+                return "ECLASS"
+        elif filename in ["ChangeLog", "metadata.xml"]:
+                return "MISC"
+        elif filename.endswith(".sh"):
+	        return "EXEC"
+        else:
+                return "OTHER"
+
+class MetaManifest(manifest.Manifest):
+
+	def getFullname(self):
+                """ Returns the absolute path to the Manifest file for this instance """
+                return os.path.join(self.pkgdir, "MetaManifest")
+
+	def sign(self):
+		'''Signs MetaManifest file with the default PORTAGE_GPG_KEY''' 
+		filename = self.getFullname()
+		portage_settings = portage.config(clone=portage.settings)
+		portage_portdbapi = portage.portdbapi(portage_settings)
+		gpgcmd = portage_settings.get("PORTAGE_GPG_SIGNING_COMMAND")
+		if gpgcmd in [None, '']:
+			raise portage.exception.MissingParameter(
+				"PORTAGE_GPG_SIGNING_COMMAND is unset! Is make.globals missing?")
+		if "${PORTAGE_GPG_KEY}" in gpgcmd and "PORTAGE_GPG_KEY" not in portage_settings:
+			raise portage.exception.MissingParameter("PORTAGE_GPG_KEY is unset!")
+		if "${PORTAGE_GPG_DIR}" in gpgcmd:
+			if "PORTAGE_GPG_DIR" not in portage_settings:
```

Comment:
Note that portage.config expose a `dict` like interface so you can just use:
```python
if "${PORTAGE_GPG_DIR}" in gpgcmd:
    portage_settings['PORTAGE_GPG_DIR'] = os.path.expanduser(
        portage_settings.get('PORTAGE_GPG_DIR', '~/.gnupg')
    )
```
instead.